lib/node_modules/@stdlib/iter/while-each/lib/main.js

```diff
@@ -0,0 +1,161 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2023 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var setReadOnly = require( '@stdlib/utils/define-nonenumerable-read-only-property' );
+var isFunction = require( '@stdlib/assert/is-function' );
+var isIteratorLike = require( '@stdlib/assert/is-iterator-like' );
+var iteratorSymbol = require( '@stdlib/symbol/iterator' );
+var format = require( '@stdlib/string/format' );
+
+
+// MAIN //
+
+/**
+* Returns an iterator which invokes a function for each iterated value before returning the iterated value.
+*
+* @param {Iterator} iterator - input iterator
+* @param {Function} predicate - function which indicates whether to continue iterating
+* @param {Function} fcn - function to invoke
+* @param {*} [thisArg] - execution context
+* @throws {TypeError} first argument must be an iterator protocol-compliant object
+* @throws {TypeError} second argument must be a function
+* @throws {TypeError} third argument must be a function
+* @returns {Iterator} iterator
+*
+* @example
+* var randu = require( '@stdlib/random/iter/randu' );
+* var isnan = require( '@stdlib/math/base/assert/is-nan' );
+*
+* function predicate( v ) {
+*     return ( v <= 0.5 );
+* }
+*
+* function assert( v ) {
+*     if ( isnan( v ) ) {
+*         throw new Error( 'should not be NaN' );
+*     }
+* }
+*
+* var iter = iterWhileEach( randu(), predicate, assert );
```

Comment:
@Jaysukh-409 I suggest not using `randu()` here, as the doctests (i.e., `returns <number>`) will randomly fail. I suggest using the example from the README which uses `array2iterator` and a static array `[ 1, 2, 3, 4 ]` with the `predicate` function updated accordingly.